### PR TITLE
Defer loading of persp mode until after user-config

### LINKED ISSUE #3881: dotspacemacs-auto-resume-layouts interferes with hooks
After enabling `dotspacemacs-auto-resume-layouts` (running on the develop branch) I've noticed that all buffers are loaded before my `dotspacemacs/user-config` runs, which means any hooks I add in there don't fire when the buffers load. 

Is this expected / is that the wrong place to configure my hooks?

For now, I've been moving these types of hooks configuration into `custom-set-variables`, which I'd rather not have to do because I can't comment them or format them.

### PR DESCRIPTION
**Pull request by cpaulik**
This should fix #3881

Tested it with `(add-hook 'python-mode-hook 'highlight-indentation-mode)` in my `user-config`

### CODE REVIEW

**Comment by aaronjensen** [comment]
Tested and confirmed that it fixes the issues I was running into with #3881

File: layers/+spacemacs/spacemacs-layouts/packages.el [hunk 1]
```diff
@@ -112,7 +112,7 @@
       ;; re-activate during `dotspacemacs/sync-configuration-layers' - see
       ;; issues #5925 and #3875)
       (unless (bound-and-true-p persp-mode)
-        (persp-mode))
+        (spacemacs/defer-until-after-user-config #'persp-mode))
```
Comment (bmag) [inline layers/+spacemacs/spacemacs-layouts/packages.el]: I'm not sure how the deferring works here, but won't it be more robust to defer the `bound-and-true-p` guard as well?

Comment (cpaulik) [inline layers/+spacemacs/spacemacs-layouts/packages.el]: You are right. Reading the comment above actually explains it.

Comment (cpaulik) [inline layers/+spacemacs/spacemacs-layouts/packages.el]: Put the whole thing into a function right there. Hope that is OK.

**Comment by TheBB** [comment]
Let's try it. :-)

Thanks! Cherry-picked in develop. You can safely delete your branch.

**Pull request closed without merging**